simulations/simulation_runner.py

```diff
@@ -59,26 +66,28 @@ def step(self):
         if self.round % self.report_interval == self.report_interval - 1:
             self.plot_tool.plot()
 
-        self.network.advance_time()
-
-    def _generate_new_messages(self):
-        validators = self.msg_gen(self.validator_set)
-        new_messages = []
-        for validator in validators:
-            message = validator.make_new_message()
-            self.network.send_to_all(message)
-            new_messages.append(message)
-        return new_messages
+    def _advance_time(self):
+        self.round += 1
+        self.network.set_time(self.round)
+        for validator_client in self.validator_clients:
+            validator_client.set_time(self.round)
```

Comment:
Are we making assumptions here about validators having sychronized clocks?

Reply:
Ah, seen below that this is easily changed as each validator maintains a clock as well.